nrf52840-hal/src/ieee802154.rs

```diff
@@ -0,0 +1,630 @@
+//! IEEE 802.15.4 radio
+
+use core::{
+    cmp,
+    ops::{self, RangeFrom},
+    sync::atomic::{self, Ordering},
+};
+
+use crate::clocks::{Clocks, ExternalOscillator};
+use crate::pac::{
+    generic::Variant,
+    radio::{state::STATE_A, txpower::TXPOWER_A},
+    RADIO,
+};
+
+/// IEEE 802.15.4 radio
+pub struct Radio<'c> {
+    radio: RADIO,
+    // used to freeze `Clocks`
+    _clocks: &'c (),
+}
+
+/// Default Clear Channel Assessment method = Carrier sense
+pub const DEFAULT_CCA: Cca = Cca::CarrierSense;
+
+/// Default radio channel = Channel 20 (`2_450` MHz)
+pub const DEFAULT_CHANNEL: Channel = Channel::_20;
+
+/// Default TX power = 0 dBm
+pub const DEFAULT_TXPOWER: TxPower = TxPower::_0dBm;
+
+/// Default Start of Frame Delimiter = `0xA7` (IEEE compliant)
+pub const DEFAULT_SFD: u8 = 0xA7;
+
+// TODO expose the other variants in `pac::CCAMODE_A`
+/// Clear Channel Assessment method
+pub enum Cca {
+    /// Carrier sense
+    CarrierSense,
+}
+
+/// IEEE 802.15.4 channels
+///
+/// NOTE these are NOT the same as WiFi 2.4 GHz channels
+// TODO it is possible to use non-standard frequencies below 2_400 MHz; should those be exposed too?
```

Comment:
This list is correct. To be compliant with IEEE 802.15.4 other frequencies shall not be exposed. The hardware in selected nRF devices is capable of using other frequencies, but it should not be used in 802.15.4 protocol.